### PR TITLE
feat: add `Listing live streaming broadcasts` support

### PR DESCRIPTION
**Pull request by ihors-livestorm**
This PR adds a missing client implementation for [ Listing live streaming broadcasts](https://tokbox.com/developer/rest/#list_broadcasts) support. 
The solution is heavily based on the `List archives` codebase.

The added specs are disabled right now, since I don't have any test credentials to generate the test mocks.
Any insights or feedbacks on making this Pull Request are welcomed.

### CODE REVIEW

**Comment by superchilled** [comment]
Hi @ihors-livestorm. Thanks for opening this PR! 😃 

I've not had a chance to look through it properly yet, but will make some time early next week to do that. Once I've done so, will be happy to work with you on this.

**Comment by ihors-livestorm** [comment]
@superchilled cool! sounds good to me. 👍 
Thanks for your help!

**Comment by superchilled** [comment]
Hi @ihors-livestorm. I've had a quick look at the PR. Looks good so far. I still need to pull down the branch, run the specs, and carry out a more detailed review; planning on doing this tomorrow.

Note: I've changed the base branch from `main` to `dev` (our current flow is to merge new features into `dev`, and then merge into main once we're ready to cut a new release of the gem. Switching the base branch has caused some conflicts in one of the files, which I'll need to resolve.

I'll follow up with you later this week once I've completed the above steps, etc.

File: spec/opentok/broadcasts_spec.rb
Comment (superchilled) [inline spec/opentok/broadcasts_spec.rb:261]: For all of these tests, `all` should be called on `broadcast` (singular) not `broadcasts` (plural). See [line 19](https://github.com/opentok/OpenTok-Ruby-SDK/blob/142ef849cba11715a31b3513e8d87eec376de793/spec/opentok/broadcasts_spec.rb#L19) where the variable is defined:

```ruby
let(:broadcast) { opentok.broadcasts }
```

File: lib/opentok/token_generator.rb
Comment (superchilled) [reply lib/opentok/token_generator.rb:9]: In fact this fix doesn't completely work. It fixes things when using >= 7, but causes a different issue when using < 7.

I've implemented an alternative fix in https://github.com/opentok/OpenTok-Ruby-SDK/pull/238

Comment (ihors-livestorm) [reply lib/opentok/token_generator.rb]: Ok, removed my fix for now 👍

**Comment by ihors-livestorm** [comment]
thanks for you review. I've made some changes according to your comments. 👍 
Waiting for #238 being merged, and updates for the VCR recordings. 😊

File: lib/opentok/token_generator.rb
Comment (superchilled) [reply lib/opentok/token_generator.rb]: 👍🏾

**Comment by superchilled** [comment]
Hi @ihors-livestorm! I've now pushed commits for the VCR recordings (and also to add a `require` that was missing). Also, https://github.com/opentok/OpenTok-Ruby-SDK/pull/238 has now been merged into `dev` and I have rebased from `dev`. All commits now showing in [this branch](https://github.com/opentok/OpenTok-Ruby-SDK/tree/feat/add-broadcasts-list); can you pull into your fork so that they show in the PR?

Once this is done. CI tests should hopefully pass 🤞🏾. We can then get a final review from @jeffswartz and merge the PR into `dev`.

File: lib/opentok/version.rb
Comment (ihors-livestorm) [inline lib/opentok/version.rb:3]: @superchilled I'm worried a bit if this part is fine. Looks like version downgrade right now

**Comment by ihors-livestorm** [comment]
Looks like CI tests are passing right now 🟢

File: lib/opentok/version.rb
Comment (superchilled) [reply lib/opentok/version.rb:3]: Current version is 4.2.0 in `main`, see: https://github.com/opentok/OpenTok-Ruby-SDK/blob/main/lib/opentok/version.rb

The new version *will* be 4.3.0, but will bump this in a separate PR (this current PR will be merged into `dev`, then when we are ready to release the new version of the Gem, will merge `dev` into `main` with a version bump).

**Comment by superchilled** [comment]
All checks have passed! ✅ 🙌🏾

File: lib/opentok/version.rb
Comment (ihors-livestorm) [reply lib/opentok/version.rb:3]: got it, thanks 👍

**Review by jeffswartz — APPROVED** [review]

**Review by superchilled — APPROVED** [review]
LGTM.

**Comment by superchilled** [comment]
@ihors-livestorm this is now merged into `dev`. Planning to release a new version of the Gem either later today, or tomorrow.

**Comment by superchilled** [comment]
Hi @ihors-livestorm. Version 4.3.0 of the gem (which includes this feature) has now been released: https://rubygems.org/gems/opentok/versions/4.3.0

Thank-you for all your work on this!

**Comment by ihors-livestorm** [comment]
@superchilled awesome, thanks for the update! 👍 
I'm super excited! 😍

**Comment by superchilled** [comment]
Thanks @ihors-livestorm! Let me know how you get on or if you have any issues.

**Pull request merged**
